### PR TITLE
[WIP] Update eslint-config-expensify and fix files

### PR DESCRIPTION
**Pull request by marcaaron**
### Fixed Issues

$ GH_LINK

### Tested On

- [ ] Web
- [ ] Mobile Web
- [ ] Desktop
- [ ] iOS
- [ ] Android

### CODE REVIEW

File: eslint-local-rules/no-negated-variables/index.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+function isNegatedVariableName(name) {
+    if (!name) {
+        return;
+    }
+
+    return name.includes('Not')
+      || name.includes('isNot')
+      || name.includes('cannot')
+      || name.includes('shouldNot')
+      || name.includes('cant');
```
Comment (tgolen) [inline eslint-local-rules/no-negated-variables/index.js:14]: Can also add `dont` to this list.

File: eslint-local-rules/no-onyx-in-views/no-onyx-in-views.test.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
+const RuleTester = require('eslint').RuleTester;
```
Comment (tgolen) [inline eslint-local-rules/no-onyx-in-views/no-onyx-in-views.test.js:1]: I'd love to see one that also prevents `API` from being called in views.

File: eslint-local-rules/utils.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+/**
+ * @param {String} filename
+ * @returns {Boolean}
+ */
+function isReactViewFile(filename) {
+    return filename.includes('/src/pages/') || filename.includes('/src/components');
```
Comment (tgolen) [inline eslint-local-rules/utils.js:6]: I wonder if it would be better or more consistent to only allow Onyx to be referenced from `src/lib/actions`? 🤔

**Review by tgolen — COMMENTED** [review]
LOVE LOVE LOVE This!

I'm not a big fan of the name "local-rules", but we can riff on that to see if there is a better name. Like, I wonder if we should just make these part of our eslint package?

**Comment by marcaaron** [comment]
>I'm not a big fan of the name "local-rules", but we can riff on that to see if there is a better name. Like, I wonder if we should just make these part of our eslint package?

That's a good question. The "local-rules" thing actually comes directly from the package that enables us to add them here instead of a plugin so that's not something we can change, but there are other packages that do the same thing without this restriction.

Couple thoughts about adding these to the eslint package:

- Could that present a barrier to contributors for adding rules? I think no, because it's already a public repo.
- Do we want to enable these same rules everywhere? I think it would be fine and we could always turn them off in places where it doesn't make sense or add a default config options like `['legacy', 'new', 'all']`

File: eslint-local-rules/utils.js (hunk 3 above)
Comment (marcaaron) [reply eslint-local-rules/utils.js:6]: Good point. We do say that "actions are the only thing that should set data". So probably a good thing to enforce.

This rule is currently just looking for "Onyx" usages but we could split this into:

- `set()`, `merge()`, `mergeCollection()` etc should always be called from `/lib/actions` files
- `Onyx.connect()` should only appear in `/libs/**`

**Comment by tgolen** [comment]
> Could that present a barrier to contributors for adding rules? I think no, because it's already a public repo.

Yeah, I think that since it's a public repo, it would allow them to help us even further so I think it's OK AFAIK.

> Do we want to enable these same rules everywhere?

I was thinking about this too. I think that we do. Most of these are things we'd like to be consistent with everywhere, and some of them will only apply to our code patterns in `App`.

File: eslint-local-rules/utils.js (hunk 3 above)
Comment (tgolen) [reply eslint-local-rules/utils.js:6]: Yeah, I like that. There are quite a few usages of `Onyx.connect()` outside of `actions` which seems like it is OK.

**Comment by botify** [comment]
Hey, I noticed you changed some webpack configuration files. This can break production builds. Did you remember to run a production build locally to verify they still work?

**Pull request closed without merging**